server/metrics/metrics.go

```diff
@@ -166,6 +168,21 @@ func New(config *libconfig.MetricsConfig) *Metrics {
 			},
 			[]string{"name"},
 		),
+		metricProcessesDefunct: prometheus.NewGaugeFunc(
+			prometheus.GaugeOpts{
+				Subsystem: collectors.Subsystem,
+				Name:      collectors.ProcessesDefunct.String(),
+				Help:      "Total number of defunct processes in the node",
+			},
+			func() float64 {
+				total, err := process.DefunctProcesses()
+				if err == nil {
+					return float64(total)
+				}
+				logrus.Warnf("%v", err)
```

Comment:
@saschagrunert since I am logging the error here, I'm not logging the error before returning from `DefunctProcesses()` in lines https://github.com/raisaat/cri-o/blob/monitor-defunct/internal/process/defunct_processes.go#L30 and https://github.com/raisaat/cri-o/blob/monitor-defunct/internal/process/defunct_processes.go#L36. Otherwise, I believe there would be redundant logs.

Reply:
Looks good to me, maybe just
```suggestion
				logrus.Warn(err)
```